### PR TITLE
[Test] Add tests for lib/util/

### LINKED ISSUE #143: Create integration tests for modules in lib/util
Order of completion:    
 - [x] json_encoder.py 
 - [x] graph.py
 - [x] preprocess.py
 - [x] union_find.py
 - [ ] layout.py

### PR DESCRIPTION
**Pull request by hrishikesh-dahiya**
## Description
 - Add integration test for `graph.py` and `preprocess.py` in lib/util/
 - FIxes #143 
 - Slight correction in `PULL_REQUEST_TEMPLATE.md`

## Motivation
 - To increase code coverage
 
 ## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checks
 - [x] Squash your commits down to one, or a few, discreet changesets before submitting a pull request
 - [x] Read the [contributing guide](https://github.com/DeveloperCAP/MLCAT/blob/development/.github/CONTRIBUTING.md)
 - [x] PR should be motivated, i.e. what does it fix, why, and if relevant how
 - [ ] If possible, include an example in the description, that could help all readers (including project members) to get a better picture of the change
 - [x] Avoid other runtime dependencies unless absolutely necessary 
 - [x] Rebase your commit history so that each commit is meaningful and states the intention behind the change 
 - [x] Ensure that your pull request follows the [style guidelines](https://github.com/DeveloperCAP/MLCAT/wiki/Coding-Guidelines)
 - [x] At least one commit should mention `Fixes #143` _if relevant_

## Notify reviewers
@prasadtalasila

### CODE REVIEW

File: lib/util/graph.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def get_current_leaf_nodes(list1, list2):
     return list3
 
 
-def get_leaf_nodes(write_to_file=True):
+def get_leaf_nodes(src_file, dest_file):
     """
     This function is used to compute the message-ids of leaf nodes in the thread graph.
```
Comment (prasadtalasila) [inline lib/util/graph.py:24]: @hrishikesh-dahiya The documentation needs update.

**Review by prasadtalasila — CHANGES_REQUESTED** [review]
@hrishikesh-dahiya Only the documentation change is required. The rest looks good.

**Comment by prasadtalasila** [comment]
@hrishikesh-dahiya 
There is a warning in [travis log](https://travis-ci.org/DeveloperCAP/MLCAT/jobs/339816895#L627). Please check the reason for the same.

File: lib/util/preprocess.py [hunk 2]
```diff
@@ -4,20 +4,21 @@
...
 
 	.. note:: Reference: Boris Iglewicz and David Hoaglin (1993), "Volume 16: How to Detect and Handle Outliers", The ASQC Basic References in Quality Control:Statistical Techniques, Edward F. Mykytka, Ph.D., Editor.
 	"""
 	if len(points.shape) == 1:
-		points = points[:,None]
-	median = np.median(points, axis=0)
-	diff = np.sum((points - median)**2, axis=-1)
-	diff = np.sqrt(diff)
-	med_abs_deviation = np.median(diff)
+		return np.array([False], np.bool)
```
Comment (prasadtalasila) [inline lib/util/preprocess.py:15]: @hrishikesh-dahiya 
The new code does not seem equivalent to old one. Why does it need to change?

Comment (hrishikesh-dahiya) [reply lib/util/preprocess.py:15]: @prasadtalasila med_abs_deviation becomes zero when we pass a single element array. Also, outlier has no meaning for a single point.

File: lib/util/graph.py (hunk 1 above)
Comment (hrishikesh-dahiya) [reply lib/util/graph.py:24]: documentation updated

**Pull request merged**